### DISCUSSION TITLE
Mapping purchases to user accounts

### DISCUSSION THREAD

**Discussion by maciekish** [imdhemy/laravel-in-app-purchases#422] [Q&A] [answered]
Hi,
Since it's the store which sends puchases directly to the server, how do i map puchases to users on the server where [laravel-in-app-purchases](https://github.com/imdhemy/laravel-in-app-purchases) is running? Do i have to upload the transaction ID from the mobile device to the server and map it based on that or is there a better way?

**Comment by imdhemy** [comment, marked as answer]
Yes you need to upload the receipt from the mobile app first. You can refer to this discussion 

https://github.com/imdhemy/laravel-in-app-purchases/discussions/24

**Reply by maciekish** [reply]
How about adding an [appAccountToken](https://developer.apple.com/documentation/storekit/product/purchaseoption/3749440-appaccounttoken) to the purchase request? Will this be visible in the transaction notification from Apple? My users already have uniquie device ID's that an account could be looked up from.
